### ISSUE TITLE
Invoke-Item and Start-Process freeze the terminal on Unix-like platforms (macOS, Linux)

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#10150]
# Steps to reproduce

```powershell
# Ditto for Start-Process
PSonUnix> Invoke-Item /
```

# Expected behavior

The platform's GUI file manager should launch asynchronously and the invoking terminal should continue to be usable.

# Actual behavior

The GUI file manager launches, but the invoking terminal becomes unusable - no further keyboard input is accepted.

# Environment data

```none
PowerShell Core 7.0.0-preview.1 on macOS 10.14.5
PowerShell Core 7.0.0-preview.1 on Ubuntu 18.04.2 LTS
```

**Comment by iSazonov** [comment]
For reference #9970.

**Comment by SteveL-MSFT** [comment]
This appears to be a regression in CoreFx.  We haven't made any changes in this area.  https://github.com/dotnet/corefx/issues/40193

**Comment by maxwellb** [comment]
Ditto for `Update-Help -Online`

(7.0.0-preview.2 snap)

**Comment by SteveL-MSFT** [comment]
dotnet team says this is fixed in 3.0 Preview.9 which will come out next month I think, so if that fix works, this should be fixed in PS7 Preview.4

**Comment by maxwellb** [comment]
@SteveL-MSFT I just tested the "frameworkdependent" release of powershell-7.0.0-preview.2 with the nightly 3.0.100-preview9-013948 of dotnet core. Looks good to me.
